jina/drivers/__init__.py

```diff
@@ -331,7 +331,10 @@ def pre_traverse(_docs, recur_on, context_doc=None,
             while working_docs[0].adjacency < self._adjacency_start:
                 _temp = []
                 for working_doc in working_docs:
-                    _temp.extend(working_doc.matches)
+                    if working_doc.matches:
+                        _temp.extend(working_doc.matches)
+                if not _temp:
+                    break
```

Comment:
Sorry for not catching this beforehand, but this is wrong. If you do this, you will change the chunks of the root document, even thought you define `adjacency_range=(1,1)`, which explicitly states, that the root document's chunks should not be touched.